packages/back-end/src/integrations/BigQuery.ts

```diff
@@ -78,4 +80,43 @@ export default class BigQuery extends SqlIntegration {
   castUserDateCol(column: string): string {
     return `CAST(${column} as DATETIME)`;
   }
+
+  async getInformationSchema(): Promise<InformationSchema[] | null> {
+    const projectId = this.params.projectId;
+
+    if (!projectId) {
+      return null;
+    }
+
+    const datasets = await this.runQuery(
+      `SELECT * FROM ${projectId}.INFORMATION_SCHEMA.SCHEMATA;`
+    );
+
+    const combinedResults: RawInformationSchema[] = [];
+
+    const query: string[] = [];
+
+    for (const dataset of datasets) {
+      query.push(`SELECT
+        table_name,
+        "${projectId}" as table_catalog,
+        table_schema,
+        COUNT(column_name) as column_count
+      FROM
+        ${projectId}.${dataset.schema_name}.INFORMATION_SCHEMA.COLUMNS
+        GROUP BY table_name, table_schema`);
+    }
+
+    const queryString = query.join(" UNION ALL ");
```

Comment:
@jdorn What do you think about throwing an error or returning null with an error message here instead? Ideally, it'd be great to give some context back to the user about why we weren't able to create an informationSchema.

I'm not sure exactly how to do that, though, this this runs via a job, rather than synchronously.